src/ImageSharp/Common/Helpers/Vector128Utilities.cs

```diff
@@ -347,6 +509,944 @@ public static Vector128<short> PackSignedSaturate(Vector128<int> left, Vector128
+
+        // Add
+        Vector128<short> sumLo = leftLo + rightLo;
+        Vector128<short> sumHi = leftHi + rightHi;
+
+        // Clamp to signed 8-bit range
+        Vector128<short> min = Vector128.Create((short)sbyte.MinValue);
+        Vector128<short> max = Vector128.Create((short)sbyte.MaxValue);
+
+        sumLo = Clamp(sumLo, min, max);
+        sumHi = Clamp(sumHi, min, max);
+
+        // Narrow back to signed bytes
+        return Vector128.Narrow(sumLo, sumHi);
+    }
+
+    /// <summary>
+    /// Subtract packed signed 8-bit integers in <paramref name="right"/> from packed signed 8-bit integers
+    /// in <paramref name="left"/> using saturation, and store the results.
+    /// </summary>
+    /// <param name="left">
+    /// The first vector containing packed signed 8-bit integers to subtract from.
+    /// </param>
+    /// <param name="right">
+    /// The second vector containing packed signed 8-bit integers to subtract.
+    /// </param>
+    /// <returns>
+    /// A vector containing the results of subtracting packed signed 8-bit integers
+    /// </returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Vector128<sbyte> SubtractSaturate(Vector128<sbyte> left, Vector128<sbyte> right)
+    {
+        if (Sse2.IsSupported)
+        {
+            return Sse2.SubtractSaturate(left, right);
+        }
+
+        if (AdvSimd.IsSupported)
+        {
+            return AdvSimd.SubtractSaturate(left, right);
+        }
+
+        if (PackedSimd.IsSupported)
+        {
+            return PackedSimd.SubtractSaturate(left, right);
+        }
+
+        // Widen inputs to 16-bit
+        (Vector128<short> leftLo, Vector128<short> leftHi) = Vector128.Widen(left);
+        (Vector128<short> rightLo, Vector128<short> rightHi) = Vector128.Widen(right);
+
+        // Subtract
+        Vector128<short> diffLo = leftLo - rightLo;
+        Vector128<short> diffHi = leftHi - rightHi;
+
+        // Clamp to signed 8-bit range
+        Vector128<short> min = Vector128.Create((short)sbyte.MinValue);
+        Vector128<short> max = Vector128.Create((short)sbyte.MaxValue);
+
+        diffLo = Clamp(diffLo, min, max);
+        diffHi = Clamp(diffHi, min, max);
+
+        // Narrow back to signed bytes
+        return Vector128.Narrow(diffLo, diffHi);
+    }
+
+    /// <summary>
+    /// Create mask from the most significant bit of each 8-bit element in <paramref name="value"/>, and store the result.
+    /// </summary>
+    /// <param name="value">
+    /// The vector containing packed 8-bit integers from which to create the mask.
+    /// </param>
+    /// <returns>
+    /// A 16-bit integer mask where each bit corresponds to the most significant bit of each 8-bit element
+    /// in <paramref name="value"/>.
+    /// </returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static int MoveMask(Vector128<byte> value)
```

Comment:
Pains me to say this, seeing this nice port, but `Vector128.ExtractMostSignificantBits` is equivalent to `MoveMask` (and that JIT intrinsic can benefit from AVX512 masking).